editor/components/block-switcher/test/index.js

```diff
@@ -0,0 +1,287 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+
+/**
+ * WordPress dependencies
+ */
+import { keycodes } from '@wordpress/utils';
+
+/**
+ * Internal dependencies
+ */
+import { BlockSwitcher, mapStateToProps, mapDispatchToProps } from '../';
+
+const { DOWN } = keycodes;
+
+describe( 'BlockSwitcher', () => {
+	test( 'Test block switcher without blocks', () => {
+		expect( shallow( <BlockSwitcher /> ) ).toMatchSnapshot();
+	} );
+	test( 'Test block switcher with blocks', () => {
+		const block = {
+			attributes: {
+				content: [ 'How are you?' ],
+				nodeName: 'H2',
+			},
+			isValid: true,
+			name: 'core/heading',
+			originalContent: '<h2>How are you?</h2>',
+			uid: 'a1303fd6-3e60-4fff-a770-0e0ea656c5b9',
+		};
+
+		const blocks = [
+			block,
+		];
+
+		expect( shallow( <BlockSwitcher blocks={ blocks } /> ) ).toMatchSnapshot();
+	} );
+
+	test( 'Test block switcher with multi block of different types.', () => {
+		const block1 = {
```

Comment:
Can we move those blocks level up and reuse in other tests to avoid code duplication?

Reply:
I would have to double check, but I am pretty sure that would work.